File: docs/api-endpoints.ts
export const API_BASE_URL = "http://3.26.10.0:8080";

export const API = {
  auth: {
    login: "/auth/login",
    register: "/auth/register",
    google: "/auth/google",
    apple: "/auth/apple",
    checkEmail: "/auth/check-email",
    checkUsername: "/auth/check-username",
    forgotPassword: "/auth/forgot-password",
    resetPassword: "/auth/reset-password",
    verify: "/auth/verify",
    logout: "/auth/logout",
    me: "/auth/me",
  },

  doctor: {
    dashboard: "/doctor/dashboard",
  },

  patients: {
    list: "/patients",
    search: "/patients/search",
    byId: (id: number | string) => `/patients/${id}`,
    history: (id: number | string) => `/patients/${id}/history`,
    currentVitals: (id: number | string) => `/patients/${id}/vitals/current`,
    diagnoses: (id: number | string) => `/patients/${id}/diagnoses`,
    prescriptions: (id: number | string) => `/patients/${id}/prescriptions`,
  },

  diagnoses: {
    list: "/diagnoses",
    byPatient: (patientId: number | string) => `/diagnoses/patient/${patientId}`,
    byId: (id: number | string) => `/diagnoses/${id}`,
    status: (id: number | string) => `/diagnoses/${id}/status`,
  },

  prescriptions: {
    list: "/prescriptions",
    byId: (id: number | string) => `/prescriptions/${id}`,
    byPatient: (patientId: number | string) => `/prescriptions/patient/${patientId}`,
    status: (id: number | string) => `/prescriptions/${id}/status`,
    pdf: (id: number | string) => `/prescriptions/${id}/pdf`,
  },

  nurse: {
    dashboard: "/nurse/dashboard",
    wardPatients: "/nurse/ward-patients",
    notes: "/nurse/notes",
  },

  vitals: {
    list: "/vitals",
    byId: (id: number | string) => `/vitals/${id}`,
    byPatient: (patientId: number | string) => `/vitals/patient/${patientId}`,
  },

  queue: {
    list: "/queue",
    byId: (id: number | string) => `/queue/${id}`,
  },

  tasks: {
    list: "/tasks",
    byId: (id: number | string) => `/tasks/${id}`,
  },

  billing: {
    patientSearch: "/billing/patients/search",
    insuranceProviders: "/insurance/providers",
    invoices: "/billing/invoices",
    invoiceById: (id: number | string) => `/billing/invoices/${id}`,
    invoiceItems: (id: number | string) => `/billing/invoices/${id}/items`,
    invoiceItemById: (id: number | string, itemId: number | string) =>
      `/billing/invoices/${id}/items/${itemId}`,
    invoiceDraft: (id: number | string) => `/billing/invoices/${id}/draft`,
    invoiceSend: (id: number | string) => `/billing/invoices/${id}/send`,
    invoicePdf: (id: number | string) => `/billing/invoices/${id}/pdf`,
  },

  health: {
    test: "/test",
  },
};

File: docs/api_endpoints.dart
class ApiConfig {
  static const String baseUrl = "http://3.26.10.0:8080";
}

class ApiEndpoints {
  // Auth
  static const String login = "/auth/login";
  static const String register = "/auth/register";
  static const String google = "/auth/google";
  static const String apple = "/auth/apple";
  static const String checkEmail = "/auth/check-email";
  static const String checkUsername = "/auth/check-username";
  static const String forgotPassword = "/auth/forgot-password";
  static const String resetPassword = "/auth/reset-password";
  static const String verifyToken = "/auth/verify";
  static const String logout = "/auth/logout";
  static const String me = "/auth/me";

  // Doctor Dashboard
  static const String doctorDashboard = "/doctor/dashboard";

  // Patients
  static const String patients = "/patients";
  static String patientById(int id) => "/patients/$id";
  static String patientHistory(int id) => "/patients/$id/history";
  static String patientVitalsCurrent(int id) => "/patients/$id/vitals/current";
  static String patientDiagnoses(int id) => "/patients/$id/diagnoses";
  static String patientPrescriptions(int id) => "/patients/$id/prescriptions";
  static const String patientsSearch = "/patients/search";

  // Diagnoses
  static const String diagnoses = "/diagnoses";
  static String diagnosesByPatient(int patientId) => "/diagnoses/patient/$patientId";
  static String diagnosisById(int id) => "/diagnoses/$id";
  static String diagnosisStatus(int id) => "/diagnoses/$id/status";

  // Prescriptions
  static const String prescriptions = "/prescriptions";
  static String prescriptionById(int id) => "/prescriptions/$id";
  static String prescriptionsByPatient(int patientId) => "/prescriptions/patient/$patientId";
  static String prescriptionStatus(int id) => "/prescriptions/$id/status";
  static String prescriptionPdf(int id) => "/prescriptions/$id/pdf";

  // Nurse
  static const String nurseDashboard = "/nurse/dashboard";
  static const String nurseWardPatients = "/nurse/ward-patients";
  static const String nurseNotes = "/nurse/notes";

  // Vitals
  static const String vitals = "/vitals";
  static String vitalsByPatient(int patientId) => "/vitals/patient/$patientId";
  static String vitalsById(int id) => "/vitals/$id";

  // Queue
  static const String queue = "/queue";
  static String queueById(int id) => "/queue/$id";

  // Tasks
  static const String tasks = "/tasks";
  static String taskById(int id) => "/tasks/$id";

  // Billing
  static const String billingPatientsSearch = "/billing/patients/search";
  static const String insuranceProviders = "/insurance/providers";
  static const String billingInvoices = "/billing/invoices";
  static String billingInvoiceById(int id) => "/billing/invoices/$id";
  static String billingInvoiceItems(int id) => "/billing/invoices/$id/items";
  static String billingInvoiceItemById(int id, int itemId) => "/billing/invoices/$id/items/$itemId";
  static String billingInvoiceDraft(int id) => "/billing/invoices/$id/draft";
  static String billingInvoiceSend(int id) => "/billing/invoices/$id/send";
  static String billingInvoicePdf(int id) => "/billing/invoices/$id/pdf";

  // Health
  static const String test = "/test";
}

